code/datums/mind.dm

```diff
@@ -762,7 +762,7 @@
 				return TRUE
 
 /mob/proc/sync_mind()
-	mind_initialize()	//updates the mind (or creates and initializes one if one doesn't exist)
+	mind_initialize(skip_initialization = mind)	//updates the mind (or creates and initializes one if one doesn't exist)
```

Comment:
Why do we need to call the initialize proc if we are skipping initialization? Just set the mind key and don't call the initialize proc

Reply:
I think I can say it's similar to init code has `mapload` parameter to skip some behavior.

Reply:
We shouldnt be calling this if the mind is already initialised, just directly put the key in the mind and mob

Reply:
hmm, then it looks we should have a proc `after_mind_sync()` to take some behaviour from mind_init proc.